pybamm/simulation.py

```diff
@@ -16,84 +16,109 @@ class Simulation(object):
+    discretisation : :class:`pybamm.discretisation.Mesh` instance
+       The discretisation to be used for the simulation.
     solver : :class:`pybamm.solver.Solver` instance
-        The algorithm for solving the model.
+       The algorithm for solving the model.
     name : string, optional
-        The simulation name.
+       The simulation name.
+
+    Examples
+    --------
+    >>> import pybamm
+    >>> model = pybamm.lead_acid.LOQS()
+    >>> sim = pybamm.Simulation(model)
+    >>> sim.run()
 
     """
 
-    def __init__(self, model, param=None, mesh=None, solver=None, name="unnamed"):
-        # Defaults
-        if param is None:
-            param = pybamm.ParameterValues()
-        if mesh is None:
-            mesh = pybamm.Mesh(param)
+    def __init__(
+        self,
+        model,
+        parameter_values=None,
+        discretisation=None,
+        solver=None,
+        name="unnamed",
+    ):
+        # Read defaults from model
+        if parameter_values is None:
+            parameter_values = model.default_parameter_values
+        if discretisation is None:
+            discretisation = model.default_discretisation
         if solver is None:
-            solver = pybamm.Solver()
+            solver = model.default_solver
 
         # Assign attributes
         self.model = model
-        self.param = param
-        self.mesh = mesh
+        self.parameter_values = parameter_values
+        self.discretisation = discretisation
         self.solver = solver
         self.name = name
 
-        # Initialise simulation to prepare for solving
-        # Set mesh dependent parameters
-        self.param.set_mesh(self.mesh)
-
-        # Create operators from solver
-        self.operators = self.solver.operators(self.mesh)
-
-        # Assign param, operators and mesh as model attributes
-        self.model.set_simulation(self.param, self.operators, self.mesh)
+        # Default evaluation time
+        self.default_t_eval = np.linspace(0, 1)
 
     def __str__(self):
         return self.name
 
-    def run(self, use_force=False):
-        """
-        Run the simulation.
-
-        Parameters
-        ----------
-        use_force : boolean, optional
-            If False (default), use a stored solution (if it exists). Otherwise, run the
-            model.
-        """
-        # if not use_force and os.path.isfile(self.filename):
-        #     self.load()
-        # else:
-        self.vars = self.solver.get_simulation_vars(self)
-        # self.save()
-
-    def average(self):
-        """Average simulation variable attributes over the relevant (sub)domain."""
-        self.vars.average()
-
-    @property
-    def filename_io(self):
-        path = os.path.join("out", "simulations")
-        filename = "model_{}_solver_{}".format(
-            pickle.dumps(self.model), pickle.dumps(self.solver)
-        )
-        return os.path.join(path, filename)
+    def set_parameters(self):
```

Comment:
maybe mark these as private functions i.e. `_set_parameters(self)`? I would expect that only people new to pybamm would use this class, and you don't want them calling:
```python
simulation.discretise()
simulation.set_parameters()
simulation.solve()
```